File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # approach: hashmap of num : index, indices must be different
        
        hashmap = {}
        
        for i, n in enumerate(nums):
            hashmap[n] = i
        
        for i, n in enumerate(nums):
            diff = target - n
            
            if diff in hashmap and i != hashmap[diff]:
                return [i, hashmap[diff]]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        
        for (int i = 0; i < nums.size(); i++) {
            hashmap[nums[i]] = i;
        }
        
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            
            if (hashmap.count(diff) && hashmap.at(diff) != i) {
                return {i, hashmap.at(diff)}; 
            }
        }
        
        return {-1, -1};
    }
};